tests/core/doc-discovery.test.ts

```
@@ -0,0 +1,287 @@
+    discovery = new DocDiscovery();
+    jest.clearAllMocks();
+    
+    // Setup path.join mock
+    mockPath.join.mockImplementation((...args: string[]) => args.join('/'));
+    mockPath.basename.mockImplementation((filePath: string) => filePath.split('/').pop() || '');
+  });
+
+  describe('getFilePattern', () => {
+    beforeEach(() => {
+      // Clear environment variables
+      delete process.env.DOT_AI_DOC_PATTERN;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)